src/core/services/interimElement/interimElementBase.js

```diff
@@ -0,0 +1,217 @@
+angular
+  .module('material.core')
+  .factory('$$interimElementBase', InterimElementBase);
+
+function InterimElementBase($mdUtil, $rootScope, $animate, $mdCompiler, $q, $$q, $mdTheming, $rootElement, $document,
+                            $timeout) {
+  return function(options) {
+    var API;
+    var element;
+    var showAction = $q.when(true);
+
+    prepareInterimElement();
+
+    return API = {
+      show: createAndShowElement,
+      remove: _runRemoveExecution,
+      options: options,
+      deferred: $q.defer()
+    };
+
+    function _runRemoveExecution(response, isCancelled, opts) {
+      // Abort and reject when the previous show failed.
+      if (!element) return $q.when(false);
+
+      options = angular.extend(options, opts || {});
```

Comment:
You don't need to pass in an empty object here, `extend` won't do anything if `opts` is falsey.